lint/base_linter/node_linter.py

```diff
@@ -47,6 +47,23 @@ def smart_paths_upwards(start_dir):
     )
 
 
+def is_yarn_project(path, manifest):
+    # type: (str, Dict[str, Any]) -> bool
+    package_manager = manifest.get('packageManager')
+    if isinstance(package_manager, str):
+        # When this field was being adopted and hadn't been standardised yet
+        # package managers could have scoped names, i.e. `@<scope>/<package>`
+        tokens = package_manager.split('@')
+        name = '@' + tokens[1] if package_manager.startswith('@') else tokens[0]
```

Comment:
That looks rather complicated but I also don't exactly know if we're expecting any variations here.  E.g. this 

```suggestion
        name = package_manager.rsplit('@', 1)[0]
```

should be the same.  

But the really static `return package_manager.startswith("yarn") or package_manager.startswith("@yarnpkg/berry")` makes the tests green and naively readable.

But that's a nit.

Reply:
In this case it is the same, thanks!